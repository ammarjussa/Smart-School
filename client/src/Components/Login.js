import React, { Component } from 'react';
import {Route,Switch} from 'react-router'
import Auth from '../auth'
import axios from "axios"
import {Form,Button,Card,Col,Row,Modal,Spinner} from 'react-bootstrap'
import '../App.css'
import LoginNavbar from './LoginNavbar';
import Profile from '../Profile'


class Login extends Component {

  state = {
    email: '',
    password: '',
    showModal: false,
    connectionTimeout: false,
    User: '',
    facultyProfile: null,
    adminProfile: null,
  }

  handleChange = (event) => {
    const {name, value} = event.target
    this.setState({[name]: value});
  }

  sendToServer = (message,event,cb) => {
    event.preventDefault()
    if(this.state.User==="Select User...")
    {
      alert("Invalid Credentials")
          this.setState({
          email: '',
          password: '',
          User: "Admin",
        })
    }
    else{
      console.log("sent", message.email )
      this.setState({showModal: true})
      let resp;
      axios.post("/login", {
        email: message.email,
        password: message.password,
        User: message.User
      }).then((res) => {

          console.log(res.data)
          resp = res.data.success
          let Profile = res.data.Profile
          // console.log(resp)
          this.setState({showModal:false})
          cb(resp,Profile,this.state)
      })
    }
    
    
  }

  handleUser = (ev) => {
    this.setState({User: ev.target.value})
  }

  handleSubmit = (isLogin,profile,st) => {   

    console.log("handlesubmit",isLogin)
    if(isLogin===true)
    {
      Profile.updateEmail(this.state.email)
      if(this.state.User==="Admin")
      {
        this.setState({adminProfile: profile})
        Auth.login(()=>{
          this.props.history.push("/admin")
        })
      }
      else if (this.state.User==="Faculty")
      {
        this.setState({facultyProfile: profile})
        Auth.login(()=>{
          this.props.history.push("/faculty")
        })
      }
    }
    else{
      // if(st.connectionTimeout===true)
      // {
      //   alert("Connection Timeout")
      //   this.setState({
      //   email: '',
      //   password: ''
      // })
      // }
      // else{
          alert("Invalid Credentials")
      // }
      
    }
    
  }


  render() {
    return ( 
    <div>

      <Modal
        
        size="sm"
        show={this.state.showModal}
        centered
      >
        <Modal.Body className="App">
            <Spinner  animation="grow" variant="primary" />
            <Spinner animation="grow" variant="warning" />
            <Spinner animation="grow" variant="danger" />
        </Modal.Body>
      </Modal>


      <LoginNavbar />
        <div className="center">

          <Card style={{ width: '20rem'}} >
          <div className="App">
            <Card.Header>User Login</Card.Header>
            <Card.Body>
            
            <Form
            onSubmit={(e)=>this.sendToServer(this.state,e,(resp,profile) => this.handleSubmit(resp,profile))}
            >

              <Form.Group className="UserForm" as={Col}>
                {/* <Form.Label>User</Form.Label> */}
                <Form.Control 
                name="user"
                onChange={this.handleUser}
                 as="select">
                  <option>Select User...</option>
                  <option value="Admin">Admin</option>
                  <option value="Faculty">Faculty</option>
                </Form.Control>
              </Form.Group>

             <Form.Group as={Col}>
               {/* <Form.Label>email:</Form.Label> */}
               <Form.Control
                required
                type="email"
                name="email"
                placeholder="Enter Email"
                value={this.state.email}
                onChange={this.handleChange}
               />
             </Form.Group >

             <Form.Group as={Col}>
               {/* <Form.Label>Password:</Form.Label> */}
               <Form.Control
                required
                type="password"
                placeholder="Enter Password"
                name="password" 
                value={this.state.password} 
                onChange={this.handleChange}
               />
             </Form.Group>

              <Button variant="outline-primary" type="submit">Login</Button>
          </Form>
          
            </Card.Body>
          </div>
          </Card> 
           
        </div>
    </div>
    );
  }
}

export default Login;
export {Login};