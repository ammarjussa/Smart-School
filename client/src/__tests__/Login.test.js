import { Login } from '../Components/Login';
import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';

const component = shallow(<Login />);

describe('Login Component', () => {
    it('Login renders properly', () => {
        const wrapper = shallow(<Login />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('Selecting no user results in an error', () => {
        const props = { User: 'Admin'} 
        const wrapper = component.find('.UserForm').childAt(0).simulate('change', {
            target: props}) ;
        // console.log(component.state())
        expect(toJson(wrapper)).toMatchSnapshot() ;
        expect(wrapper.state.User).toBe(undefined); 

    })
})