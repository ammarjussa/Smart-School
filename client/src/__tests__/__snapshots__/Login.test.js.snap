// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Login renders properly 1`] = `
<div>
  <Bootstrap(Modal)
    centered={true}
    show={false}
    size="sm"
  >
    <ModalBody
      className="App"
    >
      <Bootstrap(Spinner)
        animation="grow"
        variant="primary"
      />
      <Bootstrap(Spinner)
        animation="grow"
        variant="warning"
      />
      <Bootstrap(Spinner)
        animation="grow"
        variant="danger"
      />
    </ModalBody>
  </Bootstrap(Modal)>
  <LoginNavBar />
  <div
    className="center"
  >
    <Card
      as="div"
      body={false}
      style={
        Object {
          "width": "20rem",
        }
      }
    >
      <div
        className="App"
      >
        <CardHeader>
          User Login
        </CardHeader>
        <CardBody>
          <Form
            as="form"
            inline={false}
            onSubmit={[Function]}
          >
            <FormGroup
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "as": "div",
                  },
                  "displayName": "Col",
                  "render": [Function],
                }
              }
              className="UserForm"
            >
              <FormControl
                as="select"
                name="user"
                onChange={[Function]}
              >
                <option>
                  Select User...
                </option>
                <option
                  value="Admin"
                >
                  Admin
                </option>
                <option
                  value="Faculty"
                >
                  Faculty
                </option>
              </FormControl>
            </FormGroup>
            <FormGroup
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "as": "div",
                  },
                  "displayName": "Col",
                  "render": [Function],
                }
              }
            >
              <FormControl
                as="input"
                name="email"
                onChange={[Function]}
                placeholder="Enter Email"
                required={true}
                type="email"
                value=""
              />
            </FormGroup>
            <FormGroup
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "as": "div",
                  },
                  "displayName": "Col",
                  "render": [Function],
                }
              }
            >
              <FormControl
                as="input"
                name="password"
                onChange={[Function]}
                placeholder="Enter Password"
                required={true}
                type="password"
                value=""
              />
            </FormGroup>
            <Button
              active={false}
              disabled={false}
              type="submit"
              variant="outline-primary"
            >
              Login
            </Button>
          </Form>
        </CardBody>
      </div>
    </Card>
  </div>
</div>
`;

exports[`Login Component Selecting no user results in an error 1`] = `
<FormControl
  as="select"
  name="user"
  onChange={[Function]}
>
  <option>
    Select User...
  </option>
  <option
    value="Admin"
  >
    Admin
  </option>
  <option
    value="Faculty"
  >
    Faculty
  </option>
</FormControl>
`;
